snow/engine/common/engine.go

```diff
@@ -14,6 +15,16 @@ import (
 	"github.com/ava-labs/avalanchego/utils/set"
 )
 
+var (
+	_ error = (*AppError)(nil)
+
+	// ErrTimeout is used to signal a response timeout
+	ErrTimeout = &AppError{
+		Code:    -1,
```

Comment:
We could consider just making this `0` but I felt like it would be good for us to just reserve `0` as an undefined error in case something we don't expect comes up in the future.

Reply:
How about doing like we try and do with enums (see [uber guidelines](https://github.com/uber-go/guide/blob/master/style.md#start-enums-at-one)) and explictly define an "undefined/unknown" error to use as default value?